### ISSUE TITLE
Record inheritdoc shows in quick info but still get CS1573

### ISSUE THREAD

**Issue by Sergio0694** [dotnet/roslyn#67326]
**Version Used**: 4.5.0-6.23123.11 (271ccd71)

**Steps to Reproduce**:

Paste the following:

```csharp
/// <summary>
/// A simple foo.
/// </summary>
/// <param name="Name">A name.</param>
public record Foo(string Name)
{
}

/// <summary>
/// A simple bar.
/// </summary>
/// <param name="Stuff">Some stuff</param>
/// <inheritdoc cref="Foo"/>
public record Bar(string Name, string Stuff) : Foo(Name);
```

And set these in the .csproj:

```xml
<AnalysisLevelStyle>7-all</AnalysisLevelStyle>
<GenerateDocumentationFile>true</GenerateDocumentationFile>
```

**Expected Behavior**:

No warning should be displayed. All items are documented or they're inheriting stuff.

**Actual Behavior**:

I get this:

![](https://user-images.githubusercontent.com/10199417/225487039-157dc5d3-f8cf-4779-839d-f27089ec99ce.png)

That is, quick info _does_ show the inherited docs from the base record type, but then I still get a CS1573 saying "Name" is missing XML docs. Since I'm using `inheritdoc` here, I'd assume it'd mean "use the inherited stuff to fill in any missing docs from the XML docs above this inheritdoc". Which seems to be what's happening: "Bar" and "Stuff" are correctly showing XML docs in quick info and they show no warnings, and "Name" is showing the quick info docs inherited from the base type (and also I'm not getting CS1591, which I would get if that was outright missing docs. But then why does CS1573 show up? 🤔

cc. @RikkiGibson

**Comment by RikkiGibson** [comment]
It would be useful to know if the documentation file contains the inherited docs in all expected places, e.g. in the type symbol, as well as primary constructor and synthesized properties.

**Comment by Sergio0694** [comment]
I'm seeing this in all kinds of members, not just records:

![](https://github.com/dotnet/roslyn/assets/10199417/823740b3-19e4-4ae2-b64c-9cddb2075076)

Eg. here I'm inheriting everything and manually adding the missing XML doc for that extra parameter that's not also in the other type. Quick info works just fine everywhere, but I still get these warnings on all params. This just makes no sense to me, as if Quick info is working when hovering over everything, to me that seems to indicate the XML docs _are_ there and working fine? 🤔

The workaround is to just manually copy all docs, which is extremely verbose and feels like a massive chore.

FYI @sharwell and @jcouv, potentially related to #69126? I mean if this is by design why is Quick info working then?

**Comment by sharwell** [comment]
@Sergio0694 My recommendation is to either disable CS1573, or reverse the use of `<inheritdoc>` so the full documentation appears on the member with the most complete set of parameters (other members that use `<inheritdoc>` may inherit extra parameters, but there is no warning for that case).

**Comment by smaillet** [comment]
@sharwell - That recommendation is not always possible. Sometimes there is an `<inheritdoc/>` tag on a method that has different parameters. The same summary and remarks should be inherited and any of the same parameters but additional ones need docs. (E.g. it has some but not all of the params + some distinct ones) This is where, things get into the unexpected as the presumption, from the user perspective, is that an `<inheritdoc/>` tag actually inherits everything applicable. If I instead have to specify each part, then it's more pain than it's worth.
